gnmi/fakedevice/gobgp.go

```diff
@@ -0,0 +1,383 @@
+							newSessionState := telemetry.E_Bgp_Neighbor_SessionState(enumCode)
+							if neigh.SessionState != newSessionState {
+								log.V(1).Infof("Peer %s transitioned to session state %s", ps.NeighborAddress, v.Name)
+								neigh.SessionState = newSessionState
+							}
+							found = true
+							break
+						}
+					}
+				}
+				if !found {
+					log.Warningf("Unknown neighbor session-state value received: %v", ps.SessionState)
+				}
+			} else {
+				log.Warning("Received peer update from an unknown peer:", ps.NeighborAddress)
+			}
+			if success := updateAppliedConfig(prevApplied, false); !success {
+				log.Errorf("goBgpTask: updating applied configuration failed")
+			}
+		}
+	}); err != nil {
+		log.Fatal(err)
+	}
+
+	bgpPath := configpath.DeviceRoot("").NetworkInstance("default").Protocol(config.PolicyTypes_INSTALL_PROTOCOL_TYPE_BGP, "BGP").Bgp()
+	asPaths, _, err := ygot.ResolvePath(bgpPath.Global().As())
+	if err != nil {
+		log.Errorf("goBgpTask failed to initialize due to error: %v", err)
+	}
+	routeIDPaths, _, err := ygot.ResolvePath(bgpPath.Global().RouterId())
+	if err != nil {
+		log.Errorf("goBgpTask failed to initialize due to error: %v", err)
+	}
+	peerAsPaths, _, err := ygot.ResolvePath(bgpPath.NeighborAny().PeerAs())
+	if err != nil {
+		log.Errorf("goBgpTask failed to initialize due to error: %v", err)
+	}
+	neighAddrPaths, _, err := ygot.ResolvePath(bgpPath.NeighborAny().NeighborAddress())
+	if err != nil {
+		log.Errorf("goBgpTask failed to initialize due to error: %v", err)
+	}
+
+	var global api.Global
+	global.ListenPort = listenPort
+
+	go func() {
+		defer remove()
+		for {
+			item, _, err := q.Next(context.Background())
+			if coalesce.IsClosedQueue(err) {
+				return
+			}
+			n, ok := item.(*ctree.Leaf)
+			if !ok || n == nil {
+				log.Errorf("goBgpTask invalid cache node: %#v", item)
+				return
+			}
+			v := n.Value()
+			no, ok := v.(*gpb.Notification)
+			if !ok || no == nil {
+				log.Errorf("goBgpTask invalid cache node, expected non-nil *gpb.Notification type, got: %#v", v)
+				return
+			}
+
+			// Update the view of the intended config.
+			// Note: We're guaranteed that whatever update came from the collector is valid since we validate before storing.
+			shouldProcess := false
+			for _, u := range no.Update {
+				switch {
+				case matchingPath(u.Path, asPaths), matchingPath(u.Path, routeIDPaths), matchingPath(u.Path, neighAddrPaths), matchingPath(u.Path, peerAsPaths):
+					log.V(1).Infof("Received update path: %s", prototext.Format(u))
+					shouldProcess = true
+				default:
+					log.V(1).Infof("goBgpTask: update path received isn't matched by any handlers: %s", prototext.Format(u.Path))
+				}
+			}
+			for _, u := range no.Delete {
+				log.V(1).Infof("Received delete path: %s", prototext.Format(u))
+				switch {
+				case len(u.Elem) > 0:
+				case len(u.Element) > 0: //nolint:staticcheck //lint:ignore SA1019 Specifically handling deprecated gNMI Element fields.
```

Comment:
i think we can not support about deprecated fields. we should return an error in RPCs. TODO is fine

Reply:
Done